File: skills/clawhub/real-browser-qa-ceki-main/examples/scraping.py
"""Scraping a site with anti-bot protection using a real browser."""
import asyncio

from ceki_browser import Browser


async def main():
    async with Browser(token="YOUR_TOKEN") as br:
        async with await br.session(
            mode="incognito",
            domain_hints=["news.ycombinator.com"],
            geo="US",
        ) as s:
            await s.navigate("https://news.ycombinator.com")

            items = await s.query_all(
                "a.titlelink",
                attributes=["textContent", "href"],
                limit=30,
            )
            for el in items.elements:
                print(f"{el.get('textContent')} — {el.get('href')}")

            page_html = await s.get_html("body", outer=False)
            print(f"\nPage HTML size: {len(page_html.html)} chars")

            screenshot = await s.screenshot(format="png")
            print(f"Screenshot: {len(screenshot.data)} base64 chars")


if __name__ == "__main__":
    asyncio.run(main())

File: skills/clawhub/real-browser-qa-ceki-main/examples/scraping.ts
import { Browser } from '@ceki/sdk';

const br = new Browser({ token: 'YOUR_TOKEN' });
await br.connect();

const s = await br.openSession({
  mode: 'incognito',
  domainHints: ['news.ycombinator.com'],
  geo: 'US',
});

await s.navigate('https://news.ycombinator.com');

const items = await s.queryAll('a.titlelink', ['textContent', 'href'], 30);
for (const el of items.elements) {
  console.log(`${el.textContent} — ${el.href}`);
}

const pageHtml = await s.getHtml('body', false);
console.log(`\nPage HTML size: ${pageHtml.html.length} chars`);

const screenshot = await s.screenshot('png');
console.log(`Screenshot: ${screenshot.data.length} base64 chars`);

await s.close();
await br.close();
